src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -4191,11 +4148,50 @@ DataPartStoragePtr StorageReplicatedMergeTree::fetchExistsPart(
     return part->data_part_storage;
 }
 
+std::optional<bool> StorageReplicatedMergeTree::hasMetadataInZooKeeper() const
+{
+    {
+        std::lock_guard lock(initialization_mutex);
+        if (init_phase == InitializationPhase::INITIALIZING)
+            throw Exception(ErrorCodes::NOT_INITIALIZED, "Table {} is not initialized yet", getStorageID().getFullTableName());
+    }
+
+    return has_metadata_in_zookeeper;
```

Comment:
that's true, but I'm more interested what to do in case it's nullopt and DROP is called. 
Are we okay with dropping everything locally while not being sure if there is any metadata in ZK?

Reply:
maybe we can control it with a setting 🤔

Reply:
Currently DROP tries to remove metadata from ZK if it's not sure, so it will throw an exception if ZK is not available. Seems like it will work this way without additional changes and exception in `hasMetadataInZooKeeper` is redundant

Reply:
Ahhh, I wasn't reading the check correctly, you're right. Sorry!